tests/scripts/thread-cert/Cert_5_5_05_SplitMergeREED.py

```diff
@@ -175,11 +183,71 @@ def test(self):
         self.assertEqual(self.nodes[ROUTER1].get_state(), 'child')
         self.assertEqual(self.nodes[REED1].get_state(), 'router')
 
+        self.collect_ipaddrs()
+
         addrs = self.nodes[LEADER].get_addrs()
         for addr in addrs:
             if addr[0:4] != 'fe80':
                 self.assertTrue(self.nodes[ROUTER1].ping(addr))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_2 = pv.vars['ROUTER_2']
+        ROUTER_3 = pv.vars['ROUTER_3']
+        REED = pv.vars['REED']
+        router1_pkts = pkts.filter_wpan_src64(ROUTER_1)
+        router2_pkts = pkts.filter_wpan_src64(ROUTER_2)
+        reed_pkts = pkts.filter_wpan_src64(REED)
+
+        router2_pkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next()
+        _start_idx = router2_pkts.index
+        _reed_pkts = reed_pkts.range(_start_idx)
+
+        # Step 3: Remove Router_3 from the network
+        # Router_1 attempt to re-attach to partition.
+        # Sends multicast Parent Request to Routers and REEDs
+        router1_pkts.range(_start_idx).must_next()
+        router1_pkts.filter_mle_cmd(MLE_PARENT_REQUEST).must_next()
+        _end_idx = router1_pkts.index
+
+        # Step 2: The DUT MUST NOT attempt to become an active router by sending an Address Solicit Request
+        reed_pkts.copy().range(
+            _start_idx, _end_idx).filter_coap_request("/a/as").must_not_next()
+
+        # Step 4: The DUT send a Parent Response to Router_1
+        _reed_pkts.filter_mle_cmd(MLE_PARENT_RESPONSE).must_next().must_verify(
+            lambda p: p.wpan.dst64 == ROUTER_1)
+
+        # Step 5: Router_1 Send MLE Child ID Request to the DUT
+        router1_pkts.range(
+            _reed_pkts.index).filter_mle_cmd(MLE_CHILD_ID_REQUEST).must_next()
+
+        # Step 6: DUT send an Address Solicit Request to Leader,
+        # receives short address and becomes a router
+        _reed_pkts.filter_coap_request("/a/as").must_next().must_verify(
+            lambda p: p.coap.tlv.ext_mac_addr and p.coap.tlv.status)
+
+        # Step 7: DUT send a Multicast Link Request
+        _reed_pkts.filter_mle_cmd(MLE_LINK_REQUEST).must_next().must_verify(
+            lambda p: {18, 13, 0, 11, 3, 16} <= set(p.mle.tlv.type))
+
+        # Step 8: DUT send Child ID Response to Router_1
+        _reed_pkts.filter_mle_cmd(MLE_CHILD_ID_RESPONSE).must_next(
+        ).must_verify(lambda p: p.wpan.dst64 == ROUTER_1)
+
+        # Step 9: The DUT MUST route the ICMPv6 Echo request to the Leader.
+        # The DUT MUST route the ICMPv6 Echo reply back to Router_1
+        leader_mleid = pv.vars['LEADER_MLEID']
+        router1_mleid = pv.vars['ROUTER_1_MLEID']
+        _reed_pkts.filter(lambda p: p.ipv6.src == router1_mleid and p.ipv6.dst
```

Comment:
Add a new filter function? Seems it's often used.
```suggestion
        _reed_pkts.filter_ipv6_src_dst(router1_mleid, leader_mleid)
```